packages/flutter/lib/src/cupertino/expansion_tile.dart

```diff
@@ -0,0 +1,230 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport 'package:flutter/material.dart';
+///
+/// @docImport 'list_section.dart';
+library;
+
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'icons.dart';
+import 'list_tile.dart';
+import 'theme.dart';
+
+/// The curve of the animation used to expand or collapse the
+/// [CupertinoExpansionTile].
+///
+/// Eyeballed from an iPhone 15 simulator running iOS 17.5.
+const Curve _kAnimationCurve = Curves.easeInOut;
+
+/// The duration of the animation used to expand or collapse the
+/// [CupertinoExpansionTile].
+///
+/// Eyeballed from an iPhone 15 simulator running iOS 17.5.
+const Duration _kAnimationDuration = Duration(milliseconds: 250);
+
+/// The font size of the rotating trailing icon in the header of a
+/// [CupertinoExpansionTile].
+///
+/// Eyeballed from an iPhone 15 simulator running iOS 17.5.
+const double _kIconFontSize = 15.0;
+
+/// The height of the header in a [CupertinoExpansionTile], which is the default
+/// [CupertinoListTile].
+const double _kHeaderHeight = 44.0;
+
+/// Defines how a [CupertinoExpansionTile] should transition its child between
+/// its collapsed state and its expanded state.
+enum ExpansionTileTransitionMode {
+  /// Transition by fading a fully extended [CupertinoExpansionTile.child].
+  ///
+  /// When the [CupertinoExpansionTile] expands, the child appears fully extended
+  /// and fades into view. When the [CupertinoExpansionTile] collapses, the child
+  /// remains fully extended and fades out of view.
+  fade,
+
+  /// Transition by scrolling [CupertinoExpansionTile.child] under the header.
+  ///
+  /// When the [CupertinoExpansionTile] expands, the child scrolls from under the
+  /// header until it becomes fully extended. When the [CupertinoExpansionTile]
+  /// collapses, the child scrolls under the header until it is fully collapsed.
+  scroll,
+}
+
+/// A single-line [CupertinoListTile] with an expansion arrow icon that expands
+/// or collapses the tile to reveal or hide the [child].
+///
+/// See also:
+///
+///  * [ExpansionTile], the Material Design equivalent.
+///  * [CupertinoListSection], useful for creating an expansion tile [child].
+///  * [CupertinoListTile], the header of a [CupertinoExpansionTile].
```

Comment:
Nit: Are there any UIKit or SwiftUI docs that should be linked to, or better to not directly tie to them?